content/2022/videookulografija.md

```diff
@@ -27,30 +40,81 @@ Kako bi lakše detektovali ivice naše zenice prvo je potrebno RGB(crvena-zelena
 10. $B4=(B1+B2+B3)/3$
 11. $I1=(R4+G4+B4+UV)/4$
 
+
 U koracima od 1 do 3 izračunavamo osvetljenost i vrednosti hrominacije naše primarne slike u boji. RGB vrednosti su aproksimirane u koracima od 5 do 10 korišćenjem RGB komponenti. U 11. koraku izračunavanjem proseka vrednosti R4, G4, B4 i UV dobijamo vrednost I1 koja predstavlja našu sliku u sivom spektru. 
 
 #### **Zamućivanje**
 Kako bi naša detekcija ivice funkcionisala, neophodno je da sliku zamutimo Gausovim filterom i na taj način smanjimo količinu detalja na datoj slici. Gausov filter je zapravo niskopropusni filter  jer ima efekat smanjenja visokofrekventnih komponenti slike. Korišćenjem Gausove funkije za dve dimenzije dobijamo vrednosti iz kojih se gradi konvoluciona matrica koja se primenjuje na sliku sivog spektra. Nove vrednosti svakog piksela postavljaju se na srednje vrednosti susednih piksela. Vrenosti originalnog piksela dobija najveću Gausovu vrednost, susedni pikseli dobijaju manje vrednosti kako bi se njihova udaljenost od origanalnih piksela poveća. Na ovaj način se rezultira zamućenost kojom se čuvaju granice i ivce bolje nego drugim metodama.
 
 
+#### **Binarizacija**
+Binarizacija za cilj ima da izdvoji objekat iz slike, u ovom slučaju zenicu. Postiže se tako što sliku koju smo prebacili u sivu spektar sada konvertujemo u crno-belo. Prvo je potrebno naći trešhold vrednosti, odnosno vrednost praga, što radimo pomoću histograma. Kao trešhold uzimamo najveću vrednost kumulativne sume histograma. Sada dodeljujemo vrednosti pikselima na sledeći način – oni pikseli čija je vrednost manja ili jednaka od vrednosti trešholda, dobijaju vrednost 0 (postaju crni), a oni čija je vrednost veća od trešholda dobijaju vrednost 255 (postaju beli). Sada možemo da vidimo jasno izdvojenu zenicu (slika znice). 
+
 #### **Detekcija ivica**
 Kako bi detektovale ivice koristili smo funkciju Keni(eng. *Canny*). Detektor ivica Keni će piksel detektovati kao ivivcu ako je veličina gradijenta piksela veća nego veličina piksela sa obe njegove strane u pravcu promene maksimalnog inteziteta. Proces detekcije[^2] je sledeći:
 1. Prvo je potrebno sliku zamutiti Gausovim filtrom kako bi se smanjili detalji na datoj slici.
 2. Sledeći korak je da izračunamo veličinu i pravac(pravac maksimalne promene inteziteta piksela) gradijenta.
 3. Kada je veličina gradijenta piksela veća od njemu dva susedna, piksel označavamo kao ivicu. U suprotnom, piksel označavamo kao pozadinu.
 4. Uklanjanje slabih ivica pomoću praga histerza.
 
+#### **Hafova transformacija za krug**
+Hafova transformacija se može koristiti za određivanje parametara kruga kada je poznat broj tačaka koje padaju na perimetar. Krug čiji je radijus R i centar sa koordinatama (a, b) može se opisati sledećim jednačinama:
+$x = a + R cos(θ), y = b + R sin(θ)$
+Kada ugao θ prođe pun krug, tačke x i y, prate opseg kruga. Kada slika ima više krugova koji spadaju u opseg, traže se trojke parametara (R, a, b) koje će definisati krugove.
+
```

Comment:
Ovo što ste napisale za Hafovu transofmraciju za krug je previše uopšteno. Ovo je fer odgovor ako treba da odgovorite u 10 sekundi, ali morate više da istražite kako ovo funkcioniše, tj kako se ovi parametri nalaze. Podsetio bih vas da je naša metoda imala mnogo parametara, te vašim objašnjenjem rada ove transformacije treba da bude jasno zašto oni postoje